### PR TITLE
Add integration tests, parser tests and logging

### PR DESCRIPTION
**Pull request by aakanksha-rai**

### CODE REVIEW

File: src/main/java/tatracker/logic/parser/sort/SortCommandParser.java [hunk 1]
```diff
@@ -61,16 +72,20 @@ public SortCommand parse(String userInput) throws ParseException {
         switch (commandWord) {
 
         case CommandWords.SORT_ALL:
+            System.out.println(hasType);
```
Comment (PotatoCombat) [inline src/main/java/tatracker/logic/parser/sort/SortCommandParser.java]: Logging!

File: src/test/java/tatracker/testutil/TypicalTaTracker.java [hunk 2]
```diff
@@ -57,4 +57,14 @@ public static TaTracker getTypicalTaTrackerWithGroups() {
...
+    /**
+     * Returns an {@code TaTracker}.
+     */
+    public static TaTracker getTypicalTaTracker() {
+
+        TaTracker tat = new TaTracker();
+
+        return tat;
+    }
+
```
Comment (PotatoCombat) [inline src/test/java/tatracker/testutil/TypicalTaTracker.java:69]: Err?

File: src/test/java/tatracker/logic/parser/sort/SortCommandParserTest.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+    @Test
+    public void parse_allFieldsPresent_success() {
+        SortType type = SortType.ALPHABETIC;
+        String expectedModule = "CS3243";
+        String expectedGroup = "G04";
+
+
+        // whitespace only preamble
+        assertParseSuccess(parser, " group m/CS3243 g/G04 t/alpha",
```
Comment (PotatoCombat) [inline src/test/java/tatracker/logic/parser/sort/SortCommandParserTest.java:25]: No group isn't a preamble. Group is part of the full command word.
The preamble would be this: ____blanks____m/CS3243
Think of it as sort group + WHITESPACES + arguments

File: src/test/java/tatracker/logic/parser/sort/SortCommandParserTest.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                new SortGroupCommand(type, expectedGroup, expectedModule));
+
+        // different order
+        assertParseSuccess(parser, " group t/alphabetically m/CS3243 g/G04",
+                new SortGroupCommand(type, expectedGroup, expectedModule));
+
+        // whitespace only preamble
+        assertParseSuccess(parser, " module m/CS3243 t/alphabetically",
+                new SortModuleCommand(type, expectedModule));
```
Comment (PotatoCombat) [inline src/test/java/tatracker/logic/parser/sort/SortCommandParserTest.java]: Same here this doesn't have a preamble

Comment (PotatoCombat) [reply src/test/java/tatracker/logic/parser/sort/SortCommandParserTest.java]: eg. module                 m/CS3243 t/alpha

File: src/test/java/tatracker/logic/parser/sort/SortCommandParserTest.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        assertParseFailure(parser, " module t/alphabetically",
+                expectedModuleMessage);
+
+        // missing type prefix
+        assertParseFailure(parser, " module m/CS3243",
+                expectedModuleMessage);
+
+        // missing type prefix
+        assertParseFailure(parser, " all m/CS3243",
+                expectedAllMessage);
```
Comment (PotatoCombat) [inline src/test/java/tatracker/logic/parser/sort/SortCommandParserTest.java]: I thought sort all doesn't have a module prefix?

File: src/test/java/tatracker/logic/parser/sort/SortCommandParserTest.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        assertParseSuccess(parser, " module t/alphabetically m/CS3243",
+                new SortModuleCommand(type, expectedModule));
+
+        assertParseSuccess(parser, " all t/alphabetically",
+                new SortCommand(type));
+
+    }
+
+    @Test
+    public void parse_compulsoryFieldMissing_failure() {
```
Comment (PotatoCombat) [inline src/test/java/tatracker/logic/parser/sort/SortCommandParserTest.java:57]: Wld you need an invalid parts as well?

File: src/test/java/tatracker/logic/parser/sort/SortCommandParserTest.java [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+        // whitespace only preamble
+        assertParseSuccess(parser, " module m/CS3243 t/alphabetically",
+                new SortModuleCommand(type, expectedModule));
+
+        // different order
+        assertParseSuccess(parser, " module t/alphabetically m/CS3243",
+                new SortModuleCommand(type, expectedModule));
+
+        assertParseSuccess(parser, " all t/alphabetically",
```
Comment (PotatoCombat) [inline src/test/java/tatracker/logic/parser/sort/SortCommandParserTest.java:47]: how about multiple prefix, last accepted

File: src/test/java/tatracker/logic/parser/module/EditModuleCommandParserTest.java [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+        // different order
+        assertParseSuccess(parser, " n/Intro To AI m/CS3243", new EditModuleCommand(expectedModule, expectedName));
+
+        // multiple names - last name accepted
+        // different order
+        assertParseSuccess(parser, " n/Fake Name n/Intro To AI m/CS3243",
+                new EditModuleCommand(expectedModule, expectedName));
+    }
+
+    @Test
```
Comment (PotatoCombat) [inline src/test/java/tatracker/logic/parser/module/EditModuleCommandParserTest.java:32]: Need an invalid parts?

File: src/test/java/tatracker/logic/parser/module/DeleteModuleCommandParserTest.java [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+
+import tatracker.commons.core.Messages;
+import tatracker.logic.commands.module.DeleteModuleCommand;
+
+public class DeleteModuleCommandParserTest {
+
+    private DeleteModuleCommandParser parser = new DeleteModuleCommandParser();
+
+    @Test
+    public void parse_allFieldsPresent_success() {
```
Comment (PotatoCombat) [inline src/test/java/tatracker/logic/parser/module/DeleteModuleCommandParserTest.java:16]: Invalid field?

File: src/test/java/tatracker/logic/parser/module/EditModuleCommandParserTest.java (hunk 8 above)
Comment (PotatoCombat) [reply src/test/java/tatracker/logic/parser/module/EditModuleCommandParserTest.java:32]: no arguments failure -> will catch th ui freezing bug like in sort commands

File: src/test/java/tatracker/logic/parser/sort/SortCommandParserTest.java (hunk 6 above)
Comment (PotatoCombat) [reply src/test/java/tatracker/logic/parser/sort/SortCommandParserTest.java:57]: Also missing all prefixes -> catch the ui freezing bug

File: src/test/java/tatracker/logic/parser/module/AddModuleCommandParserTest.java [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+        // multiple names - last name accepted
+        // different order
+        assertParseSuccess(parser, " n/Fake Name n/Intro To AI m/CS3243", new AddModuleCommand(expectedModule));
+    }
+
+    @Test
+    public void parse_compulsoryFieldMissing_failure() {
+        String expectedMessage = Messages.getInvalidCommandMessage(AddModuleCommand.DETAILS.getUsage());
+
+        // missing name prefix
```
Comment (PotatoCombat) [inline src/test/java/tatracker/logic/parser/module/AddModuleCommandParserTest.java:37]: Missing all fields?

File: src/test/java/tatracker/logic/parser/group/EditGroupCommandParserTest.java [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+                new EditGroupCommand(expectedGroup, expectedModule, newGroup, null));
+    }
+
+    @Test
+    public void parse_compulsoryFieldMissing_failure() {
+        String expectedMessage = Messages.getInvalidCommandMessage(EditGroupCommand.DETAILS.getUsage());
+
+        // missing group prefix
+        assertParseFailure(parser, " m/CS3243 ng/G04 nt/lab",
+                expectedMessage);
```
Comment (PotatoCombat) [inline src/test/java/tatracker/logic/parser/group/EditGroupCommandParserTest.java:46]: missing all fields

Comment (PotatoCombat) [reply src/test/java/tatracker/logic/parser/group/EditGroupCommandParserTest.java:46]: invalid parts

File: src/test/java/tatracker/logic/commands/sort/SortModuleIntegrationTest.java [hunk 12]
```diff
@@ -0,0 +1,176 @@
...
+        Module validModule = new ModuleBuilder().withIdentifier("CS2103").build();
+        Group validGroupOne = new GroupBuilder().withIdentifier("G03").withStudent(BOB).withStudent(ALICE)
+                .withStudent(AMY).withStudent(DANIEL)
+                .withStudent(GEORGE).withStudent(FIONA).build();
+        Group validGroupTwo = new GroupBuilder().withIdentifier("G06").withStudent(BOB).withStudent(ALICE)
+                .withStudent(AMY).withStudent(DANIEL)
+                .withStudent(GEORGE).withStudent(FIONA).build();
+        validModule.addGroup(validGroupOne);
+        validModule.addGroup(validGroupTwo);
+        model.addModule(validModule);
```
Comment (PotatoCombat) [inline src/test/java/tatracker/logic/commands/sort/SortModuleIntegrationTest.java:94]: Next time can save this outside in the Before each, then ur test cases will be smaller!

File: src/test/java/tatracker/logic/commands/sort/SortModuleIntegrationTest.java [hunk 13]
```diff
@@ -0,0 +1,176 @@
...
+import tatracker.model.Model;
+import tatracker.model.ModelManager;
+import tatracker.model.UserPrefs;
+import tatracker.model.group.Group;
+import tatracker.model.module.Module;
+import tatracker.model.student.Student;
+import tatracker.testutil.group.GroupBuilder;
+import tatracker.testutil.module.ModuleBuilder;
+
+public class SortModuleIntegrationTest {
```
Comment (PotatoCombat) [inline src/test/java/tatracker/logic/commands/sort/SortModuleIntegrationTest.java:30]: Good job!

File: src/test/java/tatracker/logic/commands/module/EditModuleIntegrationTest.java [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+
+    private Model model;
+
+    @BeforeEach
+    public void setUp() {
+        model = new ModelManager(getTypicalTaTracker(), new UserPrefs());
+    }
+
+    @Test
+    public void execute_moduleAcceptedByModel_editSuccessful() throws Exception {
```
Comment (PotatoCombat) [inline src/test/java/tatracker/logic/commands/module/EditModuleIntegrationTest.java:28]: wrong name, there's an edit failure in here

File: src/test/java/tatracker/logic/commands/module/EditModuleIntegrationTest.java [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+        CommandResult commandResult = editModuleCommand.execute(model);
+
+        assertEquals(String.format(EditModuleCommand.MESSAGE_EDIT_MODULE_SUCCESS, validModule.getIdentifier()),
+                commandResult.getFeedbackToUser());
+        Module moduleInModel = model.getModule(validModule.getIdentifier());
+        assertEquals(newName, moduleInModel.getName());
+    }
+
+    @Test
+    public void execute_emptyModuleName_throwsCommandException() {
```
Comment (PotatoCombat) [inline src/test/java/tatracker/logic/commands/module/EditModuleIntegrationTest.java:44]: also how about empty module code?

**Review by PotatoCombat — APPROVED** [review]
Roughly looked through! It's good to merge!
Just highlighting some things that can be changed in the comments!

File: src/test/java/tatracker/logic/commands/module/EditModuleIntegrationTest.java (hunk 15 above)
Comment (PotatoCombat) [reply src/test/java/tatracker/logic/commands/module/EditModuleIntegrationTest.java:44]: as another test case?

File: src/test/java/tatracker/testutil/TypicalTaTracker.java (hunk 2 above)
Comment (aakanksha-rai) [reply src/test/java/tatracker/testutil/TypicalTaTracker.java:69]: HAHA i

- It is needed for when u want an empty TA tracker that u can manipulate urself

File: src/test/java/tatracker/logic/parser/module/DeleteModuleCommandParserTest.java (hunk 9 above)
Comment (aakanksha-rai) [reply src/test/java/tatracker/logic/parser/module/DeleteModuleCommandParserTest.java:16]: there r no invalids in module and group codes os it all works... main checks are done in the command

**Pull request merged**